fix: create new repositories during full-sync operation

PendaGTP opened the pull request:
Fixes #709 

The full-sync script now properly creates new repositories defined in the configuration instead of only updating existing ones. This fixes the different behavior between webhook-based sync and full-sync runs.

decyjphr commented on lib/settings.js:
```diff
@@ -475,19 +475,50 @@ ${this.results.reduce((x, y) => {
+    const missingRepoResults = await Promise.all(
+      repoInConfigs
+        .filter(repo => {
+          return !processedRepos.has(`${repo.owner}/${repo.name}`) || this.isRestricted(repo.name)
```
@PendaGTP this logic seems allow restricted repos because only time the `this.isRestricted` is checked only of a repo is processed. even the that logic seems to be incorrect.
```suggestion
          return !(processedRepos.has(`${repo.owner}/${repo.name}`) || this.isRestricted(repo.name))
          // This would be same as 
          // return !processedRepos.has(`${repo.owner}/${repo.name}`) && !this.isRestricted(repo.name)
```

PendaGTP replied:
@decyjphr Thanks for your review. I've pushed an update that fixes this wrong logic.

PendaGTP committed:
fix: create new repositories during full-sync operation

The full-sync script now properly creates new repositories defined in the
configuration instead of only updating existing ones.
This fixes the different behavior between webhook-based sync and full-sync runs.

decyjphr committed:
Merge branch 'main-enterprise' into fix/create-repos-during-fullsync
